upgrade_test.py

```diff
@@ -847,6 +847,20 @@ def test_generic_cluster_upgrade(self):
 
         For multi-dc upgrades, alternates upgraded nodes between dc's.
         """
+        # In case the target version >= 3.1 we need to perform test for truncate entries
+        target_upgrade_version = self.params.get('target_upgrade_version').replace('~', '-')
+        self.truncate_entries_flag = False
+        if target_upgrade_version and parse_scylla_version(target_upgrade_version) >= parse_scylla_version('3.1') and \
+                not is_enterprise(target_upgrade_version):
+            self.truncate_entries_flag = True
+
+        # Prepare keyspace and tables for truncate test
+        if self.truncate_entries_flag:
+            self.insert_rows = 10
+            self.fill_db_data_for_truncate_test(insert_rows=self.insert_rows)
+            # Let to ks_truncate complete the schema changes
+            time.sleep(120)
```

Comment:
instead of sleeping 2 minutes, couldn't we have a verification on status of schema agreement, stream or anything else that would show us that the `truncate_ks` is ready to work?

Reply:
You're right, I replaced the sleep with `wait_for_schema_agreement`. I'll rerun the job for good measure.
If it turns out well, I'll replace it in the regular upgrade test as well.